scalameter-core/src/main/scala/org/scalameter/utils/ClassPath.scala

```diff
@@ -8,51 +8,73 @@ import org.apache.commons.lang3.SystemUtils
 
 
 
-object ClassPath {
-  /** Returns the default classpath string.
+class ClassPath private (val paths: List[String]) extends Serializable {
+  /** Returns platform dependent classpath string.
    */
-  def default: String = {
-    extract(this.getClass.getClassLoader, sys.props("java.class.path"))
+  def mkString: String = {
+    val cp = paths.mkString(File.pathSeparator)
+    if (SystemUtils.IS_OS_WINDOWS) "\"" + cp + "\"" else cp
+  }
+
+  /** Prepends library to the classpath.
+   */
+  def ::(library: String) = {
+    ClassPath.validate(library)
+    new ClassPath(library :: paths)
+  }
+
+  override def equals(obj: Any): Boolean = obj match {
+    case cl: ClassPath => paths == cl.paths
+    case _ => false
+  }
+
+  override def hashCode(): Int = paths.hashCode()
+}
+
+object ClassPath {
+  private def validate(elem: String) = {
+    require(!elem.contains(File.pathSeparatorChar), s"Classpath library contains illegal character: ${File.pathSeparatorChar}")
+    if (SystemUtils.IS_OS_WINDOWS) require(!elem.contains("\""), "Classpath library contains illegal character: \"")
   }
 
-  /** Returns the default classpath string, surrounded by quotes on Windows.
+  private def fromString(classPath: String): ClassPath = {
+    val cl = if (SystemUtils.IS_OS_WINDOWS) classPath.stripPrefix("\"").stripSuffix("\"") else classPath
```

Comment:
I'm not sure if you want to strip double quotes:

    scala> sys.props("sun.boot.class.path")
    res0: String = C:\Program Files\Java\jdk1.7.0_21\jre\lib\resources.jar;C:\Program Files\Java\jdk1.7.0_21\jre\lib\rt.jar;...

It's not a good pattern to require that classpath strings must be enclosed in double quotes if they are passed in on Windows - we have enough of other differences that we need to worry about.

I'd enclose the classpath string in double quotes only in cases where we are preparing a final string that needs to be passed to the command line, and even that only if really necessary - currently we solve the problem by starting a process from `Seq[String]` instead of `String`, which seems to work nicely.